### PR TITLE
Improve doc and cleanup for selection_optional argument in estimators

### PR DESCRIPTION
**Pull request by QRemy**
This PR fix the "all" option for in selection_optional argument of estimators and improve the docstring. Resolve issue #3295.

### CODE REVIEW

File: gammapy/estimators/flux.py [hunk 1]
```diff
@@ -42,12 +42,12 @@ class FluxEstimator(Estimator):
         Re-optimize other free model parameters.
     selection_optional : list of str
         Which additional quantities to estimate. Available options are:
-
+            * "all": all the optional steps are executed
             * "errn-errp": estimate asymmetric errors.
             * "ul": estimate upper limits.
             * "norm-scan": estimate fit statistic profiles.
```
Comment (AtreyeeS) [inline gammapy/estimators/flux.py:48]: It should be `scan` instead of `norm-scan`, no?

File: gammapy/estimators/flux_point.py [hunk 2]
```diff
@@ -784,12 +784,12 @@ class FluxPointsEstimator(Estimator):
         Re-optimize other free model parameters.
     selection_optional : list of str
         Which additional quantities to estimate. Available options are:
-
+            * "all": all the optional steps are executed
             * "errn-errp": estimate asymmetric errors on flux.
             * "ul": estimate upper limits.
             * "norm-scan": estimate fit statistic profiles.
```
Comment (AtreyeeS) [inline gammapy/estimators/flux_point.py:790]: same as before - `scan` instead of `norm-scan`

**Review by AtreyeeS — COMMENTED** [review]
Thanks a lot @QRemy ! I think this PR still needs to homogenise between `scan` and `norm-scan` keywords. Also, as I wrote in #3295, it would be helpful to for a user to have a proper message in `.plot_ts_profile` if `scan` is not execute - maybe searching for the column and throwing an error is useful, instead of incomprehensible message we have now

File: gammapy/estimators/flux.py (hunk 1 above)
Comment (QRemy) [reply gammapy/estimators/flux.py]: yes, fixed in the latest commit

**Comment by QRemy** [comment]
@AtreyeeS, the message I see for `FluxPoints.plot_ts_profiles()` is`"Missing columns for sed type '{}':" " {}".format(sed_type, missing)` which is quite explicit. Do you have an example to reproduce the odd error meaage you have ?

**Comment by QRemy** [comment]
I added a check for optional arguments missing in `._validate_table(table, sed_type, use_optional=False)`.

**Review by AtreyeeS — APPROVED** [review]
Thanks again @QRemy ! No further comments from my side

**Review by adonath — APPROVED** [review]
Thanks @QRemy and @AtreyeeS for the review. I have no further comments either...

**Pull request merged**